roles/accprovision/scripts/process_string_to_yaml.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/python3
+
+import json
+import yaml
+import copy
+import os
+
+def main():
+    cmap_input_path = os.getenv('CONFIGMAP_INPUT_FILE')
+    cmap = dict()
+    input_dict = dict()
+    with open(cmap_input_path, 'r') as stream:
+        cmap = yaml.safe_load(stream)
+        yaml_cmap_intermediate = cmap['data']['spec']
+        input_dict = yaml.safe_load(yaml_cmap_intermediate)
+        yaml_cmap = input_dict['acc_provision_input']
+    crd_input = os.getenv('CRDINPUT')
+    yaml_crd = yaml.safe_load(crd_input)
+
+    crd_definition = os.getenv('CRD_DEFINITION')
+    crd_definition = yaml.safe_load(crd_definition) 
+    deep_merge(yaml_crd, yaml_cmap, crd_definition)
+
+    # Write configmap file
+    input_dict['acc_provision_input'] = yaml_crd
+    cmap['data']['spec'] = json.dumps(input_dict)
+    cmap['metadata'].pop('managedFields', None)
+    cmap['metadata'].pop('annotations', None)
+    cmap['metadata'].pop('creationTimestamp', None)
+    cmap['metadata'].pop('resourceVersion', None)
+    cmap['metadata'].pop('uid', None)
```

Comment:
In metadata, there are other autogenerated fields as well to pop like "generation", selfLink . Check with the API schema supported in metadata for the crd definition. Do we need only the above things to be removed?

Reply:
These are the following fields I see in the configmap schema, do we need anything else to be removed then?:
f:metadata:
        f:labels:
          .: {}
          f:aci-containers-config-version: {}
          f:network-plugin: {}
        f:ownerReferences:
          .: {}
          k:{"uid":"e3492916-4ce2-469e-80da-1f07c89a11f7"}:
            .: {}
            f:apiVersion: {}
            f:kind: {}
            f:name: {}
            f:uid: {}

Reply:
Check it via actual configmap running on the cluster. You will get the idea. The fields are added by apiserver which are autogenerated.

Reply:
The only extra field I saw here which we should add if we want to remove autogenerated for configmap is "selfLink" .

Reply:
Added selfLink too, it's deprecated and removed in k8s1.21 but the code wont fail regardless